ldaptor/protocols/ldap/ldapsyntax.py

```diff
@@ -278,7 +278,7 @@ def __hash__(self):
         return hash(str(self))
 
     def bind(self, password):
-        r = pureldap.LDAPBindRequest(dn=str(self.dn), auth=password)
+        r = pureldap.LDAPBindRequest(dn=self.dn, auth=password)
```

Comment:
is this a change in API? from bytes to DN object?

Reply:
The same idea as in ``delta`` module above: ``LDAPBindRequest`` converts its arguments to byte strings.